library/common-library/templates/_hostnetwork.tpl

```diff
@@ -7,20 +7,42 @@ Returns "true" if `hostNetwork` is enabled, otherwise fallbacks to a overridable
 {{- /* This allows us to use `$global` as an empty dict directly in case `Values.global` does not exists */ -}}
 {{- $global := index .Values "global" | default dict -}}
 
-{{- /* `get` will return "" (empty string) if value is not found, and the value otherwise, so we can type-assert with kindIs */ -}}
+{{- /*
+`get` will return "" (empty string) if value is not found, and the value otherwise, so we can type-assert with kindIs
+
+We also want only to return when this is true, returning `false` here will template "false" (string) when doing
+an `(include "common.hostNetwork" .)`, which is not an "empty string" so it is `true` if it is used
+as an evaluation somewhere else.
+*/ -}}
 {{- if get .Values "hostNetwork" | kindIs "bool" -}}
-    {{- .Values.hostNetwork -}}
+    {{- if .Values.hostNetwork -}}
+        {{- .Values.hostNetwork -}}
+    {{- end -}}
 {{- else if get $global "hostNetwork" | kindIs "bool" -}}
-    {{- $global.hostNetwork -}}
+    {{- if $global.hostNetwork -}}
+        {{- $global.hostNetwork -}}
+    {{- end -}}
 {{- else -}}
     {{- include "common.hostNetwork.defaultOverride" . -}}
 {{- end -}}
 {{- end -}}
 
 
+{{- /*
+Abstraction of the hostNetwork toggle.
+This helper allows to override the global `.global.hostNetwork` with the value of `.hostNetwork`.
+Returns "true" if `hostNetwork` is enabled, otherwise fallbacks to a overridable function
+*/ -}}
+{{- define "common.hostNetwork.boolean" -}}
```

Comment:
ditto
```suggestion
{{- define "common.hostNetwork.bool" -}}
```

or maybe

```suggestion
{{- define "common.hostNetwork.value" -}}
```

Reply:
I like `value`. Going for it.

Reply:
Solved in [#cd27a22](https://github.com/newrelic/helm-charts/pull/756/commits/cd27a2262b8b953a7e5f921de89f8c81f3ead7c4)